driver-core/src/test/unit/com/mongodb/client/model/AggregatesSpecification.groovy

```diff
@@ -306,6 +309,15 @@ class AggregatesSpecification extends Specification {
     def 'should render $out'() {
         expect:
         toBson(out('authors')) == parse('{ $out : "authors" }')
+        toBson(out('authors', new AggregateOutStageOptions().mode(REPLACE_COLLECTION))) == parse('{ $out : "authors" }')
+        toBson(out('authors', new AggregateOutStageOptions().mode(REPLACE_DOCUMENTS))) ==
+                parse('{ $out : {mode : "replaceDocuments", to : "authors" } }')
+        toBson(out('authors', new AggregateOutStageOptions().mode(INSERT_DOCUMENTS))) ==
+                parse('{ $out : {mode : "insertDocuments", to : "authors" } }')
+        toBson(out('authors', new AggregateOutStageOptions().databaseName('db1'))) ==
+                parse('{ $out : {mode : "replaceCollection", to : "authors", db : "db1" } }')
```

Comment:
So setting the databaseName also implicitly sets the mode? Where / how does that happen?

Reply:
It's a combination of the options class, in which mode defaults to REPLACE_COLLECTION, and the conditional in OutStage.  If all the options are the default values, the old-style $out where the value is just the collection name is used.  The server treats this style as implicitly replace-collection, since that's the current behavior.  But if databaseName is specified in options, then OutStage uses the new-style $out where the value is a document.  In this style mode is required, so the default value of mode (or whatever the application may have set it to) is included.